File: gaussian_unknown_precision_jl/gaussian_unknown_precision_model.cpp
// [[Rcpp::plugins("cpp11")]]
// [[Rcpp::depends(ilike, RcppArmadillo, BH, dqrng, sitmo)]]

#include <ilike.h>

/***data,y=my_rnorm(100L)***/

/***evaluate_log_prior,my_dlnorm(parameters.tau,1,1,TRUE)***/

/***simulate_prior,tau=my_rlnorm(1L,1,1)***/

/***evaluate_log_likelihood,sum(R_interface_to_julia_dnorm(data.y,0,1/sqrt(parameters.tau),TRUE))***/

File: gaussian_unknown_precision_jl/gaussian_unknown_precision_model.R
R_interface_to_julia_dnorm = function(y,m,s,use_log=TRUE)
{
  sum = 0
  for (i in 1:length(y))
  {
    sum = sum + my_dnorm(as.double(y[i]),as.double(m),as.double(s),as.logical(use_log))
  }
  return(sum)
}
